### ISSUE TITLE
Parse edgelist bug

### ISSUE THREAD

**Issue by hcars** [networkx/networkx#4116]
When calling `parse_edgelist()` using comma delimiters, it will fail to parse correctly fro edges with multiple attributes.

Ex. of bad edge:
`1,2,{'test':1, 'test_other':2}`

`parse_edgelist()` will recognize the comma in the braces as delimiter and try to split them up.

**Comment by rossbar** [comment]
Here's a minimal reproducing example:

```python
>>> import networkx as nx
>>> edge_list = ['1, 2, {"weight": 1, "color": "green"}']
>>> G = nx.parse_edgelist(edge_list, delimiter=',')
Traceback (most recent call last)
   ...
IndentationError
   ...
TypeError: Failed to convert edge data ([' {"weight": 1', ' "color": "green"}']) to dictionary.
```

Side note: this seems like a good candidate for for `from None` instead of `from e` in the exception raising as the `IndentationError` is confusing.

**Comment by crsqq** [comment]
Proposed a fix with #4125
